R/module_teal_data.R

```diff
@@ -127,7 +128,7 @@ srv_validate_reactive_teal_data <- function(id, # nolint: object_length
       }
 
       # to handle errors and qenv.error(s)
-      if (inherits(data_out, c("qenv.error", "error"))) {
+      if (inherits(data_out, c("qenv.error"))) {
```

Comment:
Hey @vedhav I managed to reduce this to a one liner error message

<img width="668" alt="image" src="https://github.com/user-attachments/assets/9520b9c9-4c21-4d48-b6da-d9381c8d5817">


with

```r
pkgload::load_all("teal.slice")
pkgload::load_all("teal.data")
pkgload::load_all("teal")

pkgload::load_all(".")
data <- teal_data(iris = iris, mtcars = mtcars)
my_transformers <- list(
  teal_transform_module(
    label = "Dummy",
    ui = function(id) actionButton(NS(id, "submit"), "Submit"),
    server = function(id, data) {
      moduleServer(id, function(input, output, session) {
        reactive({
          within(
            data(),
            {
              # if (count %% 2 != 0) {
              #   stop("error")
              # }
              iris <- head(iris, count)
            },
            count = input$submit
          )
        })
      })
    }
  )
)
init(
  data = data,
  modules = list(
    example_module(
      transformers = my_transformers
    )
  )
) |>
  runApp()

```

Reply:
Before it gave an infinite amount or error messages added in an infinite recursive loop

Reply:
It turns out it was because of using the new reactive's name as `data`. It is resolved in this commit 17671ad984f46a187c3a2854d4f2d3ee0caeb36f